src/HybridSolver.cpp

```diff
@@ -816,48 +821,48 @@ void HybridSolver::BuildSpectralAMGePreconditioner()
 #if SMOOTHG_USE_SAAMGE
     saamge::proc_init(comm_);
 
-    mfem::Table elem_dof = MatrixToTable(Agg_multiplier_);
+    mfem::Table* elem_dof = MatrixToTable(Agg_multiplier_);
     mfem::SparseMatrix multiplier_Agg = smoothg::Transpose(Agg_multiplier_);
     mfem::SparseMatrix Agg_Agg = smoothg::Mult(Agg_multiplier_, multiplier_Agg);
     auto elem_adjacency_matrix = MetisGraphPartitioner::getAdjacency(Agg_Agg);
-    mfem::Table elem_elem = MatrixToTable(elem_adjacency_matrix);
+    mfem::Table* elem_elem = MatrixToTable(elem_adjacency_matrix);
 
     // Mark dofs that are shared by more than one processor
     saamge::SharedEntityCommunication<mfem::Vector> sec(comm_, *multiplier_d_td_);
-    std::vector<saamge::agg_dof_status_t> bdr_dofs(elem_dof.Width(), 0);
+    std::vector<saamge::agg_dof_status_t> bdr_dofs(elem_dof->Width(), 0);
     for (unsigned int i = 0; i < bdr_dofs.size(); i++)
     {
         if (sec.NumNeighbors(i) > 1)
             SA_SET_FLAGS(bdr_dofs[i], AGG_ON_PROC_IFACE_FLAG);
     }
 
-    int num_elems = elem_elem.Size();
-    std::vector<int> nparts_arr(saamge_param_->num_levels - 1);
-    nparts_arr[0] = (num_elems / saamge_param_->first_coarsen_factor) + 1;
+    int num_elems = elem_elem->Size();
+    sa_nparts_.resize(saamge_param_->num_levels - 1);
+    sa_nparts_[0] = (num_elems / saamge_param_->first_coarsen_factor) + 1;
 
     bool first_do_aggregates = (saamge_param_->num_levels <= 2 && saamge_param_->do_aggregates);
-    auto apr = saamge::agg_create_partitioning_fine(
-                   *pHybridSystem_, num_elems, &elem_dof, &elem_elem, nullptr, bdr_dofs.data(),
-                   &(nparts_arr[0]), multiplier_d_td_.get(), first_do_aggregates);
+    sa_apr_ = saamge::agg_create_partitioning_fine(
```

Comment:
I would avoid returning raw pointers if possible, especially just to satisfy an optional dependency.  Maybe just change to
 `mfem::Table* elem_dof = new Table(MatrixToTable(Agg_multiplier_));` or if that forces a copy, the swap idea works better.

Reply:
Got you, I will switch the return type back the previous setting. I think using swap is better (it's not as neat, but I do want to avoid copy one more time).